File: backend/src/main/java/com/codeloom/backend/config/TestSecurityConfig.java
package com.codeloom.backend.config;

import org.springframework.context.annotation.*;
import org.springframework.security.config.annotation.web.builders.HttpSecurity;
import org.springframework.security.web.SecurityFilterChain;

@Configuration
@Profile("test")
public class TestSecurityConfig {
  @Bean
  SecurityFilterChain securityFilterChain(HttpSecurity h) throws Exception {
    h.authorizeHttpRequests(a -> a.anyRequest().authenticated());
    return h.build();
  }
}

File: backend/src/main/kotlin/com/codeloom/backend/config/TestSecurityConfig.kt
package com.codeloom.backend.config

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Profile
import org.springframework.security.config.annotation.web.builders.HttpSecurity
import org.springframework.security.config.annotation.web.invoke
import org.springframework.security.web.SecurityFilterChain


@Configuration
@Profile("test")
class TestSecurityConfig {
    @Bean
    fun securityFilterChain(http: HttpSecurity): SecurityFilterChain {
        http.invoke {
            authorizeHttpRequests {
                authorize(anyRequest, authenticated)
            }
        }

        return http.build()
    }
}